### PR TITLE
add deep-freeze@0.0.1 w/ npm auto-update

### LINKED ISSUE #8231: [Request] deep-freeze
**Library name:** deep-freeze
**Git repository url:** https://github.com/substack/deep-freeze
**npm package url(optional):** 
**License(s):** [public domain](https://github.com/substack/deep-freeze/blob/master/LICENSE)
**Official homepage:**
**Wanna say something? Leave message here:**
# 

Notes from cdnjs maintainer:
Please read the README.md and CONTRIBUTING.md document first.

You are welcome to add a library via sending pull request,
it'll be faster then just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by sufuf3**
Pull request for issue: #8231 
Related issue(s): # #
@kennynaoh please help me review this PR, thank you.
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/substack/deep-freeze
 * Official website (optional, not the repository): 
 * NPM package url (optional): https://www.npmjs.com/package/deep-freeze
 * License and its reference: public domain
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 8
   - Count of stars: 601
   - Count of forks: 23
 * NPM download stats (optional):
   - Downloads in the last day: 1173
   - Downloads in the last week: 33965
   - Downloads in the last month: 128174

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by maruilian11** [comment]
Why not check ` Project has public repository on famous online hosting platform (or been hosted on npm)`?

**Comment by maruilian11** [comment]
don't we need `index.js`? 
Check with the author if it can use in browser if you are not sure about that.

**Comment by sufuf3** [comment]
@maruilian11 I modified the file, please help me review this PR, thank you.

**Comment by maruilian11** [comment]
Have you checked it is compatible with browser?

**Comment by sufuf3** [comment]
It can't compatible with some specific browser.
There are some specific browser issues at https://github.com/substack/deep-freeze/issues

**Comment by PeterDaveHello** [comment]
@sufuf3 That's fine, we just need to make sure it can work on browser, not just nodejs.

File: ajax/libs/deep-freeze/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "email": "mail@substack.net",
+    "url": "http://substack.net"
+  },
+  "license": "public domain",
+  "npmName": "deep-freeze",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "index.js"
```
Comment (maruilian11) [inline ajax/libs/deep-freeze/package.json]: use the rule can get minified js file and map file, too.

**Comment by sufuf3** [comment]
@maruilian11 I modified again, please help me review this PR, thank you.

File: ajax/libs/deep-freeze/package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "deep",
+    "object",
+    "recursive"
+  ],
+  "author": {
+    "name": "James Halliday",
+    "email": "mail@substack.net",
+    "url": "http://substack.net"
+  },
+  "license": "public domain",
```
Comment (maruilian11) [inline ajax/libs/deep-freeze/package.json]: https://developer.mozilla.org/en-US/docs/MDN/About#Copyrights_and_licenses
you can find SPDX License in https://spdx.org/licenses/

**Comment by sufuf3** [comment]
@maruilian11 I modified the license part, please help me review this PR, thank you.

**Comment by sufuf3** [comment]
at https://github.com/substack/deep-freeze/blob/master/LICENSE#L4
--> 
https://developer.mozilla.org/en-US/docs/MDN/About#Copyrights_and_licenses
--> 
https://creativecommons.org/publicdomain/zero/1.0/

**Comment by maruilian11** [comment]
I think this PR is fine.

**Comment by PeterDaveHello** [comment]
`Pull request for issue: #8321` -> `Pull request for issue: #8231` ...

**Comment by PeterDaveHello** [comment]
Looks like the author never claim that this library can run on browser, @sufuf3 the reference you provide actually doesn't mean anything, it's a good example of what a bad reference looks like, if somebody complains his car can not fly on the USA's sky area, does this mean it can fly on the Pacific? Maybe you can test it and tell us if it runs on browser?

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by sufuf3** [comment]
Then, I think this PR could close.

**Comment by sufuf3** [comment]
Ping @PeterDaveHello

**Comment by PeterDaveHello** [comment]
Have no idea about what are you talking about, shouldn't you confirm that? Why should I reply to your "if"? Is this the situation you need to "ping" your reviewer? I don't believe that's how you're trained to do.

**Comment by sufuf3** [comment]
Sorry about that.
I test it as the following code:
```
<html>
	<head>
		<meta charset="utf-8">
		<title></title>
	</head>
	<body>
<script type="text/javascript">
var o={
	prop:function(){},
	foo:'king'
};
o.test='this is a test';
delete o.prop;
var obj=Object.freeze(o);
console.log(o===obj);
console.log(Object.isFrozen(obj));

//the code copy from index.js start
//module.exports = function deepFreeze (o) {
function deepFreeze (o) {
  Object.freeze(o);

  Object.getOwnPropertyNames(o).forEach(function (prop) {
    if (o.hasOwnProperty(prop)
    && o[prop] !== null
    && (typeof o[prop] === "object" || typeof o[prop] === "function")
    && !Object.isFrozen(o[prop])) {
      deepFreeze(o[prop]);
    }
  });
  
  return o;
//};
}
//the code copy from index.js end
</script>
	</body>
</html>
```
and it can show in the browser of chrome console as the following:
```
true
true
```
I not good at javascript, so I use an easy way to test it.

**Comment by sufuf3** [comment]
I tried and it seems that need to use with the backend. (Learn from https://nodejs.org/api/modules.html)
So what is the next thing I should do?

**Comment by PeterDaveHello** [comment]
Discussed with its author?

**Comment by sufuf3** [comment]
I think this PR could close?

**Pull request closed without merging**